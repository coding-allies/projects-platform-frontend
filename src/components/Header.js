import React from "react";
import "../style/components/Header.css";
import { AppContext } from "../contexts/AppContext";

class Header extends React.Component {
  static contextType = AppContext;
  constructor(props) {
    super(props);
    this.state = {
      user: false
    };
    this.signUserOut = this.signUserOut.bind(this);
    this.signUserIn = this.signUserIn.bind(this);
  }

  signUserIn(e) {
    this.setState(state => ({
      user: this.context.user.name
    }));
  }

  signUserOut(e) {
    this.setState(state => ({
      user: null
    }));
  }

  render() {
    const signedOut = (
      <div className="auth">
        <button onClick={this.signUserIn}>Sign in with GitHub</button>
      </div>
    );

    const signedIn = (
      <div className="auth">
        <p>Hello, {this.state.user}</p>
        <button onClick={this.signUserOut}>Log Out</button>
      </div>
    );

    return (
      <header>
        <h1>she's collaborating</h1>
        {this.state.user ? signedIn : signedOut}
      </header>
    );
  }
}

export default Header;
