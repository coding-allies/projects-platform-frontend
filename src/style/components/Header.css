header {
  padding: 24px 32px 24px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: #FFF;
  border-bottom: 1px dotted #d1d1d3;
}

.auth {
  display: flex;
  align-items: center;
}

header h1 {
  flex-grow: 0;
}

header h1 a {
  color: #393939;
  text-decoration: none;
  margin-left: 0;
}

header > * > * {
  margin-left: 1em;
}

header nav {
  flex-grow: 1;
  align-self: auto;
}

header nav ul {
  list-style: none;
  padding: 0;
  display: flex;
}

header nav ul li {
  padding: 1em;
}

header nav ul li a {
  color: #393939;
  text-decoration: none;
  padding-bottom: 3px;
}

header nav ul li a.current {
  border-bottom: 3px solid turquoise;
}
