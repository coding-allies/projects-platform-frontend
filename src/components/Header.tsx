import * as React from "react";
import { Link, NavLink } from "react-router-dom";
import { AppContext } from "../contexts/AppContext";
import logo from "../scp-logo.svg";
import "../style/components/Header.css";

class Header extends React.Component<{}, {}> {
  static contextType = AppContext;

  constructor(props: any) {
    super(props);
    this.signUserOut = this.signUserOut.bind(this);
    this.signUserIn = this.signUserIn.bind(this);
  }

  signUserIn(e: any) {
    this.context.login();
  }

  async signUserOut(e: any) {
    this.context.logout();
  }

  render() {
    const user = this.context.user;
    const signedOut = (
      <div className="auth">
        <button onClick={this.signUserIn}>Sign in with GitHub</button>
      </div>
    );

    const signedIn = (
      <div className="auth">
        <p>Hello, {user.name}!</p>
        <Link to="/add-project" className="button-link">Add Project</Link>
        <button onClick={this.signUserOut}>Log Out</button>
      </div>
    );

    return (
      <header>
        <h1 className="logo">
          <Link to="/">
            <img src={logo} alt="She's Coding Projects" />
          </Link>
        </h1>
        <nav>
          <ul>
            <li><NavLink exact to="/" activeClassName="current">Home</NavLink></li>
            <li><NavLink to="/projects" activeClassName="current">Projects</NavLink></li>
          </ul>
        </nav>
        {user.is_authenticated ? signedIn : signedOut}
      </header>
    );
  }
}

export default Header;
