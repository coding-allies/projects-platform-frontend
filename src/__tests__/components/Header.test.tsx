import * as React from "react";
import {MemoryRouter as Router} from "react-router-dom";
import renderer from "react-test-renderer";

import Header from "../../components/Header";

it("renders Header correctly", () => {
  const component = renderer.create(<Router><Header /></Router>).toJSON();
  expect(component).toMatchSnapshot();
});
