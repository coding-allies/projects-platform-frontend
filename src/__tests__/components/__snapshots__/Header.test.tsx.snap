// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header correctly 1`] = `
<header>
  <h1
    className="logo"
  >
    <a
      href="/projects"
      onClick={[Function]}
    >
      she's collaborating
    </a>
  </h1>
  <nav>
    <ul>
      <li>
        <a
          aria-current="page"
          className="current"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/projects"
          onClick={[Function]}
        >
          Projects
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="auth"
  >
    <button
      onClick={[Function]}
    >
      Sign in with GitHub
    </button>
  </div>
</header>
`;
